### ISSUE TITLE
[ES6] const not treated as block-scoped

### ISSUE THREAD

**Issue by AgentME** [jshint/jshint#2268]
The following gives no warnings, as expected:

```javascript
// jshint esnext:true
{let a = 5;}
{let a = 5;}
```

But this does, even though const is also block-scoped:

```javascript
// jshint esnext:true
{const a = 5;}
{const a = 5;}
```

```
stdin: line 3, col 10, const 'a' has already been declared.
```

jshint v2.6.3

**Comment by lukeapage** [comment]
I worked on this yesterday and hope to have a PR in the next 24 hours (to save anyone else making a similiar effort)

**Comment by lukeapage** [comment]
duplicate of #1468

**Comment by lukeapage** [comment]
@jugglinmike this can be closed now

**Comment by jugglinmike** [comment]
Confirmed--resolved via gh-2326 / 1bb80f9a5551a639db2be0dea655c9edef7a9b8a . Expect this in the next release of JSHint (version > 2.7.0).
